### PR TITLE
Time format: 0 seconds is a number

### PR DESCRIPTION
**Pull request by kvc0**
When formatting a value, e.g., in the Stat panel, with a unit of Seconds,
Grafana chooses "NaN ns" when attempting to format 0 seconds.

0, however, is a number; and a quantity of 0 seconds is a perfectly
cromulent notion.  A quantity of 0 seconds should be presented as "0 s"
rather than "NaN ns"

```
>> Test Suites: 494 passed, 494 total
>> Tests:       6 skipped, 4061 passed, 4067 total
>> Snapshots:   179 passed, 179 total
>> Time:        167.998 s
>> Ran all test suites.

Running "no-only-tests" task

Running "no-focus-convey-tests" task

Done.
✨  Done in 227.62s.
```

**What this PR does / why we need it**:
Renders `select 0` as "0 s" when unit is Seconds  /  Because `select 0` does not return `NaN ns` when interpreted as "seconds."

**Special notes for your reviewer**:
A quick pass through every other time option shows them rendering 0 as "0 <unit>" as a reasonable person might otherwise naively assume.  S is special.

Before:
image

After:
image

### CODE REVIEW

File: packages/grafana-data/src/valueFormats/dateTimeFormatters.ts [hunk 1]
```diff
@@ -102,6 +102,9 @@ export function toSeconds(size: number, decimals?: DecimalCount, scaledDecimals?
   if (size === null) {
     return { text: '' };
   }
+  if (size === 0) {
+    return { text: '0 s' };
```
Comment (ryantxu) [inline packages/grafana-data/src/valueFormats/dateTimeFormatters.ts:106]:
```suggestion
    return { text: '0', suffix:' s', numeric:0 };
```

Maybe

Comment (WarriorOfWire) [reply packages/grafana-data/src/valueFormats/dateTimeFormatters.ts]: `FormattedValue` has no such member "numeric" but sure if you'd like the suffix separated then why not

**Comment by hugohaggmark** [comment]
Thank you for contributing @WarriorOfWire, could you add a test for this as well?

All the best

**Review by hugohaggmark — COMMENTED** [review]
Looks good, missing a test for this case.

File: packages/grafana-data/src/valueFormats/dateTimeFormatters.ts [hunk 2]
```diff
@@ -102,6 +102,9 @@ export function toSeconds(size: number, decimals?: DecimalCount, scaledDecimals?
   if (size === null) {
     return { text: '' };
   }
+  if (size === 0) {
```
Comment (hugohaggmark) [inline packages/grafana-data/src/valueFormats/dateTimeFormatters.ts:105]: nitpick on formatting, needs a new line.
```suggestion
  
  if (size === 0) {
```

**Comment by WarriorOfWire** [comment]
@hugohaggmark thanks for the review.  Here's the test suite results with 1 more test:

```
>> Test Suites: 494 passed, 494 total
>> Tests:       6 skipped, 4062 passed, 4068 total
>> Snapshots:   179 passed, 179 total
>> Time:        155.706 s
>> Ran all test suites.
```

File: packages/grafana-data/src/valueFormats/dateTimeFormatters.ts (hunk 2 above)
Comment (WarriorOfWire) [reply packages/grafana-data/src/valueFormats/dateTimeFormatters.ts:107]: added newline and comment.

**Review by ryantxu — APPROVED** [review]
LGTM - thank you!

**Comment by hugohaggmark** [comment]
Thank you for contributing @WarriorOfWire ⭐

**Pull request merged**